deeposlandia/postprocess.py

```diff
@@ -0,0 +1,440 @@
+        Size of the tiled images, in pixel
+    img_width : int
+        Original image width, in pixel
+    img_height : int
+        Original image height, in pixel
+    batch_size : int
+        Number of images passed in each inference batches
+
+    Returns
+    -------
+    numpy.array
+        Labelled version of the original image, where (i, j)-th pixel value
+    corresponds to its predicted label
+    """
+    img_height = img_height if img_height is not None else img_width
+    y_raw_preds = model.predict(images, batch_size=batch_size)
+    predicted_labels = np.argmax(y_raw_preds, axis=3)
+    full_labelled_image = fill_labelled_image(
+        predicted_labels, coordinates, tile_size, img_width, img_height
+    )
+    return full_labelled_image
+
+
+def draw_grid(data, img_width, img_height, tile_size):
+    """Draw a white grid on an original image labelled version, depending on
+    its tile splitting, *i.e.* draw vertical and horizontal lines each
+    `tile_size` pixels.
+
+    Parameters
+    ----------
+    data : numpy.array
+        Labelled version of an image
+    img_width : int
+        Original image width, in pixel
+    img_height : int
+        Original image height, in pixel
+    tile_size : int
+        Size of the tiled images, in pixel
+
+    Returns
+    -------
+    numpy.array
+        Labelled version of an image, with an explicit white grid that
+    highlights the tile splitting
+    """
+    gridded_data = data.copy()
+    for i in range(tile_size, img_width, tile_size):
+        gridded_data[:, i] = np.full([img_height, 3], 255, dtype=np.uint8)
+    for i in range(tile_size, img_height, tile_size):
+        gridded_data[i] = np.full([img_width, 3], 255, dtype=np.uint8)
+    return gridded_data
+
+
+def get_image_features(datapath, dataset, filename):
+    """Retrieve geotiff image features with GDAL
+
+    Use the `GetGeoTransform` method, that provides the following values:
+      + East/West location of Upper Left corner
+      + East/West pixel resolution
+      + 0.0
+      + North/South location of Upper Left corner
+      + 0.0
+      + North/South pixel resolution
+
+    See GDAL documentation (https://www.gdal.org/gdal_tutorial.html)
+
+    Parameters
+    ----------
+    datapath : str
+    dataset : str
+    filename : str
+        Name of the image file from which coordinates are extracted
+
+    Returns
+    -------
+    dict
+        Bounding box of the image (west, south, east, north coordinates), srid,
+    and size (in pixels)
+
+    """
+    input_folder = utils.prepare_input_folder(datapath, dataset)
+    filepath = os.path.join(
+        input_folder, "testing", "images", filename + ".tif"
+    )
+    ds = gdal.Open(filepath)
+    width = ds.RasterXSize
+    height = ds.RasterYSize
+    gt = ds.GetGeoTransform()
+    minx = gt[0]
+    miny = gt[3] + height * gt[5]
+    maxx = gt[0] + width * gt[1]
+    maxy = gt[3]
+    srid = int(ds.GetProjection().split('"')[-2])
+    ds = None
```

Comment:
maybe you can leave a comment about "why do you set None to ds"

Reply:
Added in the docstring: "A GDAL dataset is opened during the function execution. The corresponding variable is set to None at the end of the function so as to free memory."
Added as a comment at this line: "# Free memory used by the GDAL Dataset"